staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -465,6 +465,11 @@ func (r Request) finalURLTemplate() url.URL {
 	}
 	if r.namespaceSet && len(r.namespace) != 0 {
 		r.namespace = "{namespace}"
+	} else if strings.Contains(r.pathPrefix, "/namespaces/") {
+		// for dynamic client, namespace string is appended in r.pathPrefix in function
+		// makeURLSegments instead of setting r.namespaceSet and r.namespace, so the above
+		// check would not be able to replace namespace with {namespace} for dynamic client
+		r.pathPrefix, r.err = replaceNamespaceForDynamicClient(r.pathPrefix, r.baseURL.Path)
```

Comment:
This looks really messy to put here. Is there nothing the dynamic client can do to fix its usage of request, rather than embedding this parsing in request?

Reply:
@deads2k I see you suggested a fix in finalURLTemplate. Was this what you were envisioning? It seems pretty complex to me, and doesn't even fully solve the issue (name segment is still not replaced)